### ISSUE TITLE
Message from new Signal user appears without contact name (phone number instead)

### ISSUE THREAD

**Issue by OliverO2** [signalapp/Signal-iOS#3479]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
A message from new Signal user appears without contact name, although her contact is already present in the device's address book. It seems like Signal does not synchronize the contact along with the first incoming message.

### Steps to reproduce
1. Invite a user to Signal (via SMS/iMessage)
2. The new user sends her first message

**Actual result:** The message appears without the new user's name on the receiving device. Her phone number is shown instead. As a result, misleading options appear when accessing the contact details within Signal:
- An option to add a new contact appears, which will only result in creating a duplicate contact on the device.
- If the option to add the phone number to an existing contact is used instead, it will result in duplicate phone numbers on the device.

**Expected result:** For a better user experience, the new user's complete contact should be available with the first incoming message. So a single automatic contact sync should occur whenever there is an initial message from a contact unknown to Signal.

**Workaround:** These steps will synchronize contacts:
1. tap the icon to compose a new message
2. pull down the contact list

### Device info

**Device**: iPhone 6s

**iOS version**: 11.3.1

**Signal version:** 2.25.1.1

**Comment by leonardoporpora** [comment]
I had the same issue, and i solve it simply adding the prefix that every state has got, for example in Italy +39 .....
Let me know if that worked

**Comment by OliverO2** [comment]
Almost all of my contacts (including the one experiencing the issue) are stored with phone numbers in international format (`+<country code> ...`).. If it were a number format issue, I'd also expect the workaround mentioned above to fail.

**Comment by michaelkirk-signal** [comment]
@OliverO2 - do you have conversations with other contacts where the name is showing up correctly?

Can you send me a screenshot of Settings.app > Signal?

**Comment by OliverO2** [comment]
@michaelkirk-signal Yes, I have other conversations where the name is showing up correctly. I also have a number of contacts appearing in Signal without an accompanying conversation.

[Settings > Signal](https://user-images.githubusercontent.com/4660803/40776587-7be671b2-64cb-11e8-98ec-a15141bc6d16.PNG)

**Comment by michaelkirk-signal** [comment]
@OliverO2 - can you include a debug log? (Signal > Settings > Advanced > Submit Debug Log) and paste the link here.

**Comment by OliverO2** [comment]
@michaelkirk-signal I assume you want me to provide a debug log covering the whole process:
1. enable debug logging
2. invite a new user
3. wait for the new user's first message (hopefully reproducing the issue)
4. send the debug log

Correct?

**Comment by leonardoporpora** [comment]
Just enable debug logging and paste it here @OliverO2

**Comment by michaelkirk-signal** [comment]
> @michaelkirk-signal I assume you want me to provide a debug log covering the whole process:

Correct - if you didn't have debug logging enabled, nothing you'd done previously would be captured in the log.

**Comment by OliverO2** [comment]
OK, will do. Might take a few days as I need to find some ~guinea pig~ new user to cooperate with...

**Comment by theGrower** [comment]
i just activated the app on my new phone, Galaxy S9+ (SM-G965F/DS) International Version, though I'm using it in the states. The system has not been modified. I have added country code to some contacts, but they still do not show up. All that is showing up is the phone number only, which include the country code.

**Comment by michaelkirk-signal** [comment]
@theGrower - this is the Signal-iOS repository. If you're experiencing a bug with Signal-Android, could you post over there: https://github.com/signalapp/Signal-Android

**Comment by theGrower** [comment]
Oh cripes. I didn't even realize. Sorry about that, @michaelkirk-signal!

**Comment by denver-s** [comment]
It seems like recent changes made the name setting a mandatory feature and it could have fixed this issue.
